hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -204,22 +256,41 @@ def generate_novel_prefix(self, random):
         assert not self.is_exhausted
         novel_prefix = bytearray()
 
-        def append_int(n_bits, value):
-            novel_prefix.extend(int_to_bytes(value, bits_to_bytes(n_bits)))
+        BUFFER_SIZE = 8 * 1024
+
+        def draw(ir_type, kwargs, *, forced=None):
+            cd = ConjectureData(max_length=BUFFER_SIZE, prefix=b"", random=random)
+            draw_func = getattr(cd, f"draw_{ir_type}")
+            value = draw_func(**kwargs, forced=forced)
+            return (value, cd.buffer)
```

Comment:
using the trick of gluing two primitive providers together here to retrieve the buffer we need. Can probably be nicely extracted somewhere; this function is a bit of a mess currently with all the helper functions I defined.